css-paint-api/Overview.bs

```diff
@@ -180,32 +199,52 @@ called, the user agent <em>must</em> run the following steps:
             can also contains currently invalid properties for the user agent. For example
             <code>margin-bikeshed-property</code>.
 
-    5. Let |alphaValue| be the result of <a>Get</a>(|paintCtor|, "alpha").
+    6. Let |inputArguments| be the result of <a>Get</a>(|paintCtor|, "inputArguments").
+
+    7. If |inputArguments| is not undefined, and the result of <a>IsArray</a>(|inputArguments|) is
+        false, <a>throw</a> a <a>TypeError</a> and abort all these steps.
+
+        If |inputArguments| is undefined, let |inputArguments| be a new empty array.
+
+    8. Let |parsedInputArguments| be an empty array.
+
+    9. For each |item| in |inputArguments| perform the following substeps:
+
+        1. If the result of <a>Type</a>(|item|) is not String, <a>throw</a> a <a>TypeError</a> and
+            abort all these steps.
+
+        2. Let |parsedItem| be the result of parsing |item| according to the rules in
+            [[css-properties-values-api#supported-syntax-strings]]. If it fails to parse
+            <a>throw</a> a <a>TypeError</a> and abort all these steps.
 
-    6. If |alphaValue| is not undefined and the result of <a>Type</a>(|alpha|) is not Boolean,
+        3. Append |parsedItem| to |parsedInputArguments|.
+
+    10. Let |alphaValue| be the result of <a>Get</a>(|paintCtor|, "alpha").
+
+    11. If |alphaValue| is not undefined and the result of <a>Type</a>(|alpha|) is not Boolean,
         <a>throw</a> a <a>TypeError</a> and abort all these steps.
 
-    7. Let |alpha| be <code>true</code> if |alphaValue| is undefined, otherwise let it be the value
+    12. Let |alpha| be <code>true</code> if |alphaValue| is undefined, otherwise let it be the value
         of |alphaValue|.
 
         Note: Setting <code>alpha</code> is <code>false</code> allows user agents to anti-alias text
             an addition to performing "visibility" optimizations, e.g. not painting an image behind
             the paint image as the paint image is opaque.
 
-    8. If the result of <a>IsConstructor</a>(|paintCtor|) is false, <a>throw</a> a <a>TypeError</a>
+    13. If the result of <a>IsConstructor</a>(|paintCtor|) is false, <a>throw</a> a <a>TypeError</a>
         and abort all these steps.
 
-    9. Let |prototype| be the result of <a>Get</a>(|paintCtor|, "prototype").
+    14. Let |prototype| be the result of <a>Get</a>(|paintCtor|, "prototype").
 
-    10. If the result of <a>Type</a>(|prototype|) is not Object, <a>throw</a> a <a>TypeError</a> and
+    15. If the result of <a>Type</a>(|prototype|) is not Object, <a>throw</a> a <a>TypeError</a> and
         abort all these steps.
 
-    11. Let |paint| be the result of <a>Get</a>(|prototype|, "paint").
+    16. Let |paint| be the result of <a>Get</a>(|prototype|, "paint").
```

Comment:
Does this mean that it's *required* that "paint" be on the prototype, rather than on a higher prototype? Is there a more abstract operation we can use instead that just walks the proto chain? If so, we should just use that - there's no reason it shouldn't be able to show up on a higher proto, or on the instance.